src/infer_request.h

```diff
@@ -503,9 +503,10 @@ class InferenceRequest {
       TRITONSERVER_InferenceResponseCompleteFn_t response_fn,
       void* response_userp)
   {
-    response_factory_.reset(new InferenceResponseFactory(
-        model_shared_, id_, allocator, alloc_userp, response_fn, response_userp,
-        response_delegator_));
+    response_allocator_ = allocator;
```

Comment:
>   // Initialize the response factory that is to be used with any
  // responses produced for this request.
  
Comment above this function should be updated if making this change